{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "780a5c7d-0ec4-4338-9de0-16ef0ec28f4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqUtils import seq3\n",
    "import pandas as pd\n",
    "import datetime\n",
    "import re\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "54d361a3-dcf2-4e09-8281-1beadcaad927",
   "metadata": {},
   "source": [
    "# Process variants from gnomAD into relevant sequences\n",
    "\n",
    "We are interested in the structure of relevant in-population sequences \n",
    "\n",
    "We combine gnomAD info with the reference sequence from Uniprot to make our variants which we will then co-fold downstream"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "def42e64-50ae-4a9d-897c-d3e053a86498",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = {\"CYP3A4\": (\"./CYP3A4/CYP3A4_gnomAD_v4.1.0_ENSG00000160868.csv\", \"./CYP3A4/CYP3A4_P08684.fasta\"),\n",
    "        \"CYP2C9\": (\"./CYP2C9/CYP2C9_gnomAD_v4.1.0_ENSG00000138109.csv\", \"./CYP2C9/CYP2C9_P11712.fasta\"),\n",
    "        \"CYP2D6\": (\"./CYP2D6/CYP2D6_gnomAD_v4.1.0_ENSG00000100197.csv\", \"./CYP2D6/CYP2D6_P10635.fasta\"),\n",
    "        \"CYP2J2\": (\"./CYP2J2/CYP2J2_gnomAD_v4.1.0_ENSG00000134716.csv\", \"./CYP2J2/CYP2J2_P51589.fasta\"),\n",
    "        \"PXR\":    (\"./PXR/PXR_gnomAD_v4.1.0_ENSG00000144852.csv\", \"./PXR/PXR_O75469.fasta\"),\n",
    "        \"AHR\":    (\"./AHR/AHR_gnomAD_v4.1.0_ENSG00000106546.csv\", \"./AHR/AHR_P35869.fasta\")\n",
    "        \n",
    "       }"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "397a4a06-c058-4328-8ed4-4501dc79aa00",
   "metadata": {},
   "source": [
    "## VEPS\n",
    "\n",
    "We are not interested in intron or splice variants as we wish to be working with protein sequences, PLOF in these sequences is a bioinformatics not structural biology problem. \n",
    "\n",
    "We could keep stop-lost, stop_gained and inframe and frameshifts in but will assume PLOF at this stage. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84914a40-d1ea-4f98-8630-695a4d977136",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "b18c4e2a-566c-4e5c-93ea-f0196ecc8c0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "mis_and_PLOF_VEPs = [\"stop_lost\", \"stop_gained\", \"missense_variant\", \"inframe_deletion\", \"frameshift_variant\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "9dca2332-3de4-4b2d-a94a-7a67a3a526e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "mis_VEPs = [\"missense_variant\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "769eba50-67fa-42a4-9285-fe4e92dceb9c",
   "metadata": {},
   "source": [
    "# Transcripts\n",
    "\n",
    "We are not considering transcripts other than the canonical MANE transcript as they may not match the Uniprot, NCBI and EMBL-EBI sequences, we need to record their numbers here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "f0a226f8-c13a-4f51-81b1-917251dff033",
   "metadata": {},
   "outputs": [],
   "source": [
    "canonical_MANE_transcripts = {\n",
    "    \"CYP3A4\": \"ENST00000651514.1\"\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "0179cd33-e280-47ea-b2e5-d5c960264d0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_seq(path):\n",
    "    seq = SeqIO.parse(path, \"fasta\")\n",
    "\n",
    "    seqs = []\n",
    "    for r in seq.records:\n",
    "        seqs.append(r.seq)\n",
    "    assert len(seqs) == 1\n",
    "    sequence = seqs[0]\n",
    "    return sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "60ddd5e4-ddf5-4660-a9ba-42be72e4a528",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_hgvs(hgvs):\n",
    "    match = re.match(r\"p\\.([A-Za-z]+)(\\d+)([A-Za-z]+)$\", hgvs)\n",
    "    if match:\n",
    "        return match.groups()  # Returns (original_aa, position, new_aa)\n",
    "    return None  # Return None for frameshifts or other invalid cases\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "1bafaf94-6781-47c0-b87d-2ac04572f1ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "def apply_hgvs_annotation(ref_seq, hgvs_annotation):\n",
    "    data = parse_hgvs(hgvs_annotation)\n",
    "    print(f\"HGVS: {hgvs_annotation}\")\n",
    "    print(f\"parsed {data}\")\n",
    "\n",
    "    if data == None:\n",
    "        return None\n",
    "    else:\n",
    "        # unpack\n",
    "        original, hgvs_position, new = data\n",
    "    \n",
    "    # adjust for 1 indexing \n",
    "    biopython_idx = int(hgvs_position) -1\n",
    "\n",
    "    # check original matches\n",
    "    original_amino = ref_seq[biopython_idx]\n",
    "    \n",
    "    # convert to 1 letter \n",
    "    original_amino = seq3(original_amino)\n",
    "    \n",
    "    \n",
    "    print(f\"original {original}, amino {original_amino}\")\n",
    "    assert original == original_amino\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "2ea4b98a-714e-4181-b92d-b42216879a5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_data(target, gnomad_csv, reference_seq_fasta, veps, canoncial_transcript):\n",
    "    print(f\"processing {target} {gnomad_csv} {reference_seq_fasta}\")\n",
    "    \n",
    "    # grab gnomad data \n",
    "    variant_data = pd.read_csv(gnomad_csv)\n",
    "    canonical_only = variant_data[variant_data[\"Transcript\"] == canoncial_transcript]\n",
    "    sorted = canonical_only.sort_values(\"Allele Frequency\", ascending=False)\n",
    "    relevant_veps = sorted[sorted[\"VEP Annotation\"].isin(veps)]\n",
    "\n",
    "    # grab ref seq \n",
    "    ref_seq = read_seq(reference_seq_fasta)\n",
    "    print(ref_seq)\n",
    "    \n",
    "    relevant_veps[\"mutant_seq\"] = relevant_veps[\"Protein Consequence\"].apply(lambda x: apply_hgvs_annotation(ref_seq, x))\n",
    "\n",
    "    print(df)    \n",
    "\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "d2f7bbe5-3d35-4b9d-b580-7bab9f1cc440",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "processing CYP3A4 ./CYP3A4/CYP3A4_gnomAD_v4.1.0_ENSG00000160868.csv ./CYP3A4/CYP3A4_P08684.fasta\n",
      "MALIPDLAMETWLLLAVSLVLLYLYGTHSHGLFKKLGIPGPTPLPFLGNILSYHKGFCMFDMECHKKYGKVWGFYDGQQPVLAITDPDMIKTVLVKECYSVFTNRRPFGPVGFMKSAISIAEDEEWKRLRSLLSPTFTSGKLKEMVPIIAQYGDVLVRNLRREAETGKPVTLKDVFGAYSMDVITSTSFGVNIDSLNNPQDPFVENTKKLLRFDFLDPFFLSITVFPFLIPILEVLNICVFPREVTNFLRKSVKRMKESRLEDTQKHRVDFLQLMIDSQNSKETESHKALSDLELVAQSIIFIFAGYETTSSVLSFIMYELATHPDVQQKLQEEIDAVLPNKAPPTYDTVLQMEYLDMVVNETLRLFPIAMRLERVCKKDVEINGMFIPKGVVVMIPSYALHRDPKYWTEPEKFLPERFSKKNKDNIDPYIYTPFGSGPRNCIGMRFALMNMKLALIRVLQNFSFKPCKETQIPLKLSLGGLLQPEKPVVLKVESRDGTVSGA\n",
      "HGVS: p.Met445Thr\n",
      "parsed ('Met', '445', 'Thr')\n",
      "original Met, amino Met\n",
      "HGVS: p.Asp174His\n",
      "parsed ('Asp', '174', 'His')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Arg162Gln\n",
      "parsed ('Arg', '162', 'Gln')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Leu293Pro\n",
      "parsed ('Leu', '293', 'Pro')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Gly56Asp\n",
      "parsed ('Gly', '56', 'Asp')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Ser222Pro\n",
      "parsed ('Ser', '222', 'Pro')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Asn426Ser\n",
      "parsed ('Asn', '426', 'Ser')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Arg130Gln\n",
      "parsed ('Arg', '130', 'Gln')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Arg403Cys\n",
      "parsed ('Arg', '403', 'Cys')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Thr185Ser\n",
      "parsed ('Thr', '185', 'Ser')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Thr363Met\n",
      "parsed ('Thr', '363', 'Met')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Arg162Trp\n",
      "parsed ('Arg', '162', 'Trp')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Thr363Lys\n",
      "parsed ('Thr', '363', 'Lys')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Leu373Phe\n",
      "parsed ('Leu', '373', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Pro416Leu\n",
      "parsed ('Pro', '416', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Thr11Ile\n",
      "parsed ('Thr', '11', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Lys67Arg\n",
      "parsed ('Lys', '67', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Phe176Leu\n",
      "parsed ('Phe', '176', 'Leu')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Thr136Ala\n",
      "parsed ('Thr', '136', 'Ala')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Ile118Val\n",
      "parsed ('Ile', '118', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Lys168Asn\n",
      "parsed ('Lys', '168', 'Asn')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Met358Val\n",
      "parsed ('Met', '358', 'Val')\n",
      "original Met, amino Met\n",
      "HGVS: p.Pro488His\n",
      "parsed ('Pro', '488', 'His')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Val500Ile\n",
      "parsed ('Val', '500', 'Ile')\n",
      "original Val, amino Val\n",
      "HGVS: p.Pro467Ser\n",
      "parsed ('Pro', '467', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ile369Val\n",
      "parsed ('Ile', '369', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Glu262Lys\n",
      "parsed ('Glu', '262', 'Lys')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Met395Val\n",
      "parsed ('Met', '395', 'Val')\n",
      "original Met, amino Met\n",
      "HGVS: p.Asp277Asn\n",
      "parsed ('Asp', '277', 'Asn')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Ser478Cys\n",
      "parsed ('Ser', '478', 'Cys')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Tyr68Cys\n",
      "parsed ('Tyr', '68', 'Cys')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Asn198Ser\n",
      "parsed ('Asn', '198', 'Ser')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Leu3Pro\n",
      "parsed ('Leu', '3', 'Pro')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Ala370Ser\n",
      "parsed ('Ala', '370', 'Ser')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Val489Ile\n",
      "parsed ('Val', '489', 'Ile')\n",
      "original Val, amino Val\n",
      "HGVS: p.Phe60Leu\n",
      "parsed ('Phe', '60', 'Leu')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Ile427Val\n",
      "parsed ('Ile', '427', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Phe189Leu\n",
      "parsed ('Phe', '189', 'Leu')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Gln200His\n",
      "parsed ('Gln', '200', 'His')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Ile120Thr\n",
      "parsed ('Ile', '120', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Asn192Ser\n",
      "parsed ('Asn', '192', 'Ser')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Arg403His\n",
      "parsed ('Arg', '403', 'His')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Asn247Ile\n",
      "parsed ('Asn', '247', 'Ile')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Arg105Gln\n",
      "parsed ('Arg', '105', 'Gln')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Ile335Thr\n",
      "parsed ('Ile', '335', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ser131Leu\n",
      "parsed ('Ser', '131', 'Leu')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Lys96Glu\n",
      "parsed ('Lys', '96', 'Glu')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Pro218Arg\n",
      "parsed ('Pro', '218', 'Arg')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ser186Gly\n",
      "parsed ('Ser', '186', 'Gly')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Pro43Ser\n",
      "parsed ('Pro', '43', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Cys239Phe\n",
      "parsed ('Cys', '239', 'Phe')\n",
      "original Cys, amino Cys\n",
      "HGVS: p.Met145Val\n",
      "parsed ('Met', '145', 'Val')\n",
      "original Met, amino Met\n",
      "HGVS: p.Ala2Val\n",
      "parsed ('Ala', '2', 'Val')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Asp292Asn\n",
      "parsed ('Asp', '292', 'Asn')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Pro416Arg\n",
      "parsed ('Pro', '416', 'Arg')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Arg105Trp\n",
      "parsed ('Arg', '105', 'Trp')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Ser299Pro\n",
      "parsed ('Ser', '299', 'Pro')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Asp194Asn\n",
      "parsed ('Asp', '194', 'Asn')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Pro43Leu\n",
      "parsed ('Pro', '43', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Met395Ile\n",
      "parsed ('Met', '395', 'Ile')\n",
      "original Met, amino Met\n",
      "HGVS: p.Ala305Ser\n",
      "parsed ('Ala', '305', 'Ser')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Leu3Val\n",
      "parsed ('Leu', '3', 'Val')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Arg260His\n",
      "parsed ('Arg', '260', 'His')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Asp123Glu\n",
      "parsed ('Asp', '123', 'Glu')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Val296Met\n",
      "parsed ('Val', '296', 'Met')\n",
      "original Val, amino Val\n",
      "HGVS: p.Pro389Ser\n",
      "parsed ('Pro', '389', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ala370Val\n",
      "parsed ('Ala', '370', 'Val')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Asp217Asn\n",
      "parsed ('Asp', '217', 'Asn')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Tyr68His\n",
      "parsed ('Tyr', '68', 'His')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Leu129Ile\n",
      "parsed ('Leu', '129', 'Ile')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Met450Val\n",
      "parsed ('Met', '450', 'Val')\n",
      "original Met, amino Met\n",
      "HGVS: p.Asn247Asp\n",
      "parsed ('Asn', '247', 'Asp')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Val175Ile\n",
      "parsed ('Val', '175', 'Ile')\n",
      "original Val, amino Val\n",
      "HGVS: p.Lys342Asn\n",
      "parsed ('Lys', '342', 'Asn')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Val393Met\n",
      "parsed ('Val', '393', 'Met')\n",
      "original Val, amino Val\n",
      "HGVS: p.Asp336Glu\n",
      "parsed ('Asp', '336', 'Glu')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Gly481Arg\n",
      "parsed ('Gly', '481', 'Arg')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Glu125Asp\n",
      "parsed ('Glu', '125', 'Asp')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Ser437Thr\n",
      "parsed ('Ser', '437', 'Thr')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Val313Gly\n",
      "parsed ('Val', '313', 'Gly')\n",
      "original Val, amino Val\n",
      "HGVS: p.Val490Phe\n",
      "parsed ('Val', '490', 'Phe')\n",
      "original Val, amino Val\n",
      "HGVS: p.Phe189Ser\n",
      "parsed ('Phe', '189', 'Ser')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Lys66Arg\n",
      "parsed ('Lys', '66', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Thr27Ile\n",
      "parsed ('Thr', '27', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Leu36Phe\n",
      "parsed ('Leu', '36', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Val20Met\n",
      "parsed ('Val', '20', 'Met')\n",
      "original Val, amino Val\n",
      "HGVS: p.Tyr355Cys\n",
      "parsed ('Tyr', '355', 'Cys')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Thr310Met\n",
      "parsed ('Thr', '310', 'Met')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Val493Ala\n",
      "parsed ('Val', '493', 'Ala')\n",
      "original Val, amino Val\n",
      "HGVS: p.Asp292Glu\n",
      "parsed ('Asp', '292', 'Glu')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Ser315Phe\n",
      "parsed ('Ser', '315', 'Phe')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Gln472His\n",
      "parsed ('Gln', '472', 'His')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Ala297Thr\n",
      "parsed ('Ala', '297', 'Thr')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Phe463Cys\n",
      "parsed ('Phe', '463', 'Cys')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Ile431Arg\n",
      "parsed ('Ile', '431', 'Arg')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Asp277His\n",
      "parsed ('Asp', '277', 'His')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Lys342Arg\n",
      "parsed ('Lys', '342', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Val170Ile\n",
      "parsed ('Val', '170', 'Ile')\n",
      "original Val, amino Val\n",
      "HGVS: p.Arg403Pro\n",
      "parsed ('Arg', '403', 'Pro')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Leu229Phe\n",
      "parsed ('Leu', '229', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Phe220Val\n",
      "parsed ('Phe', '220', 'Val')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Tyr347Cys\n",
      "parsed ('Tyr', '347', 'Cys')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Asp428Glu\n",
      "parsed ('Asp', '428', 'Glu')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Leu22Val\n",
      "parsed ('Leu', '22', 'Val')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Ile184Thr\n",
      "parsed ('Ile', '184', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Lys378Arg\n",
      "parsed ('Lys', '378', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Arg260Cys\n",
      "parsed ('Arg', '260', 'Cys')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Tyr99His\n",
      "parsed ('Tyr', '99', 'His')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Leu21Val\n",
      "parsed ('Leu', '21', 'Val')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Asn192Lys\n",
      "parsed ('Asn', '192', 'Lys')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Asn462Lys\n",
      "parsed ('Asn', '462', 'Lys')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Phe228Leu\n",
      "parsed ('Phe', '228', 'Leu')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Gly112Arg\n",
      "parsed ('Gly', '112', 'Arg')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Ser139Gly\n",
      "parsed ('Ser', '139', 'Gly')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Gly77Ser\n",
      "parsed ('Gly', '77', 'Ser')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Gly77Val\n",
      "parsed ('Gly', '77', 'Val')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Pro429Leu\n",
      "parsed ('Pro', '429', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Pro202Leu\n",
      "parsed ('Pro', '202', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Met353Leu\n",
      "parsed ('Met', '353', 'Leu')\n",
      "original Met, amino Met\n",
      "HGVS: p.Leu229Arg\n",
      "parsed ('Leu', '229', 'Arg')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Ile232Val\n",
      "parsed ('Ile', '232', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Gly109Asp\n",
      "parsed ('Gly', '109', 'Asp')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Ser139Thr\n",
      "parsed ('Ser', '139', 'Thr')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Val81Leu\n",
      "parsed ('Val', '81', 'Leu')\n",
      "original Val, amino Val\n",
      "HGVS: p.Pro135Leu\n",
      "parsed ('Pro', '135', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Arg268Gln\n",
      "parsed ('Arg', '268', 'Gln')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Ile473Asn\n",
      "parsed ('Ile', '473', 'Asn')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ala178Val\n",
      "parsed ('Ala', '178', 'Val')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Glu122Asp\n",
      "parsed ('Glu', '122', 'Asp')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Ser464Phe\n",
      "parsed ('Ser', '464', 'Phe')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Pro147Leu\n",
      "parsed ('Pro', '147', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Met452Thr\n",
      "parsed ('Met', '452', 'Thr')\n",
      "original Met, amino Met\n",
      "HGVS: p.Gln461His\n",
      "parsed ('Gln', '461', 'His')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Glu205Gly\n",
      "parsed ('Glu', '205', 'Gly')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Cys58Arg\n",
      "parsed ('Cys', '58', 'Arg')\n",
      "original Cys, amino Cys\n",
      "HGVS: p.His54Arg\n",
      "parsed ('His', '54', 'Arg')\n",
      "original His, amino His\n",
      "HGVS: p.Cys64Ser\n",
      "parsed ('Cys', '64', 'Ser')\n",
      "original Cys, amino Cys\n",
      "HGVS: p.Met62Lys\n",
      "parsed ('Met', '62', 'Lys')\n",
      "original Met, amino Met\n",
      "HGVS: p.Cys98Tyr\n",
      "parsed ('Cys', '98', 'Tyr')\n",
      "original Cys, amino Cys\n",
      "HGVS: p.Ile50Thr\n",
      "parsed ('Ile', '50', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Gln79Arg\n",
      "parsed ('Gln', '79', 'Arg')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Asn49Asp\n",
      "parsed ('Asn', '49', 'Asp')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Pro39Leu\n",
      "parsed ('Pro', '39', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ala117Thr\n",
      "parsed ('Ala', '117', 'Thr')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Pro43Ala\n",
      "parsed ('Pro', '43', 'Ala')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ser116Asn\n",
      "parsed ('Ser', '116', 'Asn')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Thr310Lys\n",
      "parsed ('Thr', '310', 'Lys')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Glu125Asp\n",
      "parsed ('Glu', '125', 'Asp')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Ile300Val\n",
      "parsed ('Ile', '300', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ser464Thr\n",
      "parsed ('Ser', '464', 'Thr')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ile457Val\n",
      "parsed ('Ile', '457', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ser315Ala\n",
      "parsed ('Ser', '315', 'Ala')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ile457Asn\n",
      "parsed ('Ile', '457', 'Asn')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Val101Leu\n",
      "parsed ('Val', '101', 'Leu')\n",
      "original Val, amino Val\n",
      "HGVS: p.Lys67Asn\n",
      "parsed ('Lys', '67', 'Asn')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Lys421Arg\n",
      "parsed ('Lys', '421', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Val81Met\n",
      "parsed ('Val', '81', 'Met')\n",
      "original Val, amino Val\n",
      "HGVS: p.Met89Val\n",
      "parsed ('Met', '89', 'Val')\n",
      "original Met, amino Met\n",
      "HGVS: p.Ala2Thr\n",
      "parsed ('Ala', '2', 'Thr')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Ala2Asp\n",
      "parsed ('Ala', '2', 'Asp')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Gly502Glu\n",
      "parsed ('Gly', '502', 'Glu')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Asp336His\n",
      "parsed ('Asp', '336', 'His')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Gly480Glu\n",
      "parsed ('Gly', '480', 'Glu')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Gly436Arg\n",
      "parsed ('Gly', '436', 'Arg')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Leu482Phe\n",
      "parsed ('Leu', '482', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Asp357Glu\n",
      "parsed ('Asp', '357', 'Glu')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Pro485Arg\n",
      "parsed ('Pro', '485', 'Arg')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Val459Leu\n",
      "parsed ('Val', '459', 'Leu')\n",
      "original Val, amino Val\n",
      "HGVS: p.Ile148Val\n",
      "parsed ('Ile', '148', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ser188Thr\n",
      "parsed ('Ser', '188', 'Thr')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Lys453Asn\n",
      "parsed ('Lys', '453', 'Asn')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Gly109Arg\n",
      "parsed ('Gly', '109', 'Arg')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Gly109Cys\n",
      "parsed ('Gly', '109', 'Cys')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Glu258Val\n",
      "parsed ('Glu', '258', 'Val')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Lys55Gln\n",
      "parsed ('Lys', '55', 'Gln')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Trp72Leu\n",
      "parsed ('Trp', '72', 'Leu')\n",
      "original Trp, amino Trp\n",
      "HGVS: p.Phe219Leu\n",
      "parsed ('Phe', '219', 'Leu')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.His30Arg\n",
      "parsed ('His', '30', 'Arg')\n",
      "original His, amino His\n",
      "HGVS: p.Gly31Glu\n",
      "parsed ('Gly', '31', 'Glu')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Pro45Leu\n",
      "parsed ('Pro', '45', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Gln472Arg\n",
      "parsed ('Gln', '472', 'Arg')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Gly177Glu\n",
      "parsed ('Gly', '177', 'Glu')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Ala178Ser\n",
      "parsed ('Ala', '178', 'Ser')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Ser398Asn\n",
      "parsed ('Ser', '398', 'Asn')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Pro368Leu\n",
      "parsed ('Pro', '368', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Tyr179Ser\n",
      "parsed ('Tyr', '179', 'Ser')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Arg375Lys\n",
      "parsed ('Arg', '375', 'Lys')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Arg158Ser\n",
      "parsed ('Arg', '158', 'Ser')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Ser186Asn\n",
      "parsed ('Ser', '186', 'Asn')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ser311Asn\n",
      "parsed ('Ser', '311', 'Asn')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Thr185Ala\n",
      "parsed ('Thr', '185', 'Ala')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Pro345Ser\n",
      "parsed ('Pro', '345', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ile443Thr\n",
      "parsed ('Ile', '443', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Pro411Leu\n",
      "parsed ('Pro', '411', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Met450Thr\n",
      "parsed ('Met', '450', 'Thr')\n",
      "original Met, amino Met\n",
      "HGVS: p.Phe226Leu\n",
      "parsed ('Phe', '226', 'Leu')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Val225Asp\n",
      "parsed ('Val', '225', 'Asp')\n",
      "original Val, amino Val\n",
      "HGVS: p.Val225Ala\n",
      "parsed ('Val', '225', 'Ala')\n",
      "original Val, amino Val\n",
      "HGVS: p.Ile230Leu\n",
      "parsed ('Ile', '230', 'Leu')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Thr264Ile\n",
      "parsed ('Thr', '264', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Pro107Arg\n",
      "parsed ('Pro', '107', 'Arg')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ser252Thr\n",
      "parsed ('Ser', '252', 'Thr')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Arg106Lys\n",
      "parsed ('Arg', '106', 'Lys')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Ser420Gly\n",
      "parsed ('Ser', '420', 'Gly')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Asn49Thr\n",
      "parsed ('Asn', '49', 'Thr')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Phe113Ser\n",
      "parsed ('Phe', '113', 'Ser')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Ile90Thr\n",
      "parsed ('Ile', '90', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ser134Cys\n",
      "parsed ('Ser', '134', 'Cys')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ile38Val\n",
      "parsed ('Ile', '38', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ser116Cys\n",
      "parsed ('Ser', '116', 'Cys')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ile223Val\n",
      "parsed ('Ile', '223', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ile473Met\n",
      "parsed ('Ile', '473', 'Met')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Leu272Pro\n",
      "parsed ('Leu', '272', 'Pro')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Asp428Asn\n",
      "parsed ('Asp', '428', 'Asn')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Asp174Glu\n",
      "parsed ('Asp', '174', 'Glu')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Cys468Tyr\n",
      "parsed ('Cys', '468', 'Tyr')\n",
      "original Cys, amino Cys\n",
      "HGVS: p.Ile276Thr\n",
      "parsed ('Ile', '276', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Pro169Thr\n",
      "parsed ('Pro', '169', 'Thr')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Val175Asp\n",
      "parsed ('Val', '175', 'Asp')\n",
      "original Val, amino Val\n",
      "HGVS: p.Pro345Leu\n",
      "parsed ('Pro', '345', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ser18Ile\n",
      "parsed ('Ser', '18', 'Ile')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Glu124Gly\n",
      "parsed ('Glu', '124', 'Gly')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Glu470Lys\n",
      "parsed ('Glu', '470', 'Lys')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Gly167Asp\n",
      "parsed ('Gly', '167', 'Asp')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Ser495Thr\n",
      "parsed ('Ser', '495', 'Thr')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Pro467Leu\n",
      "parsed ('Pro', '467', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Thr323Pro\n",
      "parsed ('Thr', '323', 'Pro')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Val500Ala\n",
      "parsed ('Val', '500', 'Ala')\n",
      "original Val, amino Val\n",
      "HGVS: p.Pro389Leu\n",
      "parsed ('Pro', '389', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Pro169Ala\n",
      "parsed ('Pro', '169', 'Ala')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Leu172Phe\n",
      "parsed ('Leu', '172', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Ile184Val\n",
      "parsed ('Ile', '184', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Glu374Gln\n",
      "parsed ('Glu', '374', 'Gln')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Thr309Ala\n",
      "parsed ('Thr', '309', 'Ala')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Ala150Ser\n",
      "parsed ('Ala', '150', 'Ser')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Leu156Phe\n",
      "parsed ('Leu', '156', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Val191Glu\n",
      "parsed ('Val', '191', 'Glu')\n",
      "original Val, amino Val\n",
      "HGVS: p.Ala8Thr\n",
      "parsed ('Ala', '8', 'Thr')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Pro202Arg\n",
      "parsed ('Pro', '202', 'Arg')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Met371Arg\n",
      "parsed ('Met', '371', 'Arg')\n",
      "original Met, amino Met\n",
      "HGVS: p.Met9Leu\n",
      "parsed ('Met', '9', 'Leu')\n",
      "original Met, amino Met\n",
      "HGVS: p.Met9Val\n",
      "parsed ('Met', '9', 'Val')\n",
      "original Met, amino Met\n",
      "HGVS: p.Glu354Asp\n",
      "parsed ('Glu', '354', 'Asp')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Pro434Ser\n",
      "parsed ('Pro', '434', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Gly438Val\n",
      "parsed ('Gly', '438', 'Val')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Arg372Gly\n",
      "parsed ('Arg', '372', 'Gly')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Gly444Ala\n",
      "parsed ('Gly', '444', 'Ala')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Asn451Ser\n",
      "parsed ('Asn', '451', 'Ser')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Met452Leu\n",
      "parsed ('Met', '452', 'Leu')\n",
      "original Met, amino Met\n",
      "HGVS: p.Pro231Leu\n",
      "parsed ('Pro', '231', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ile230Val\n",
      "parsed ('Ile', '230', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Leu233Ile\n",
      "parsed ('Leu', '233', 'Ile')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Val235Ala\n",
      "parsed ('Val', '235', 'Ala')\n",
      "original Val, amino Val\n",
      "HGVS: p.Ile238Thr\n",
      "parsed ('Ile', '238', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Pro107Leu\n",
      "parsed ('Pro', '107', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Gly56Ser\n",
      "parsed ('Gly', '56', 'Ser')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Glu244Gln\n",
      "parsed ('Glu', '244', 'Gln')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Leu249Val\n",
      "parsed ('Leu', '249', 'Val')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Leu142Phe\n",
      "parsed ('Leu', '142', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Arg105Leu\n",
      "parsed ('Arg', '105', 'Leu')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Lys67Thr\n",
      "parsed ('Lys', '67', 'Thr')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Ser139Arg\n",
      "parsed ('Ser', '139', 'Arg')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Thr138Ser\n",
      "parsed ('Thr', '138', 'Ser')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Tyr53His\n",
      "parsed ('Tyr', '53', 'His')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Gly77Arg\n",
      "parsed ('Gly', '77', 'Arg')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Ala83Val\n",
      "parsed ('Ala', '83', 'Val')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Ala83Ser\n",
      "parsed ('Ala', '83', 'Ser')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Met114Val\n",
      "parsed ('Met', '114', 'Val')\n",
      "original Met, amino Met\n",
      "HGVS: p.Phe113Leu\n",
      "parsed ('Phe', '113', 'Leu')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Met114Ile\n",
      "parsed ('Met', '114', 'Ile')\n",
      "original Met, amino Met\n",
      "HGVS: p.Ser278Pro\n",
      "parsed ('Ser', '278', 'Pro')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Thr42Lys\n",
      "parsed ('Thr', '42', 'Lys')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Ser222Thr\n",
      "parsed ('Ser', '222', 'Thr')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Pro474His\n",
      "parsed ('Pro', '474', 'His')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Glu285Gly\n",
      "parsed ('Glu', '285', 'Gly')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Asp428His\n",
      "parsed ('Asp', '428', 'His')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Lys127Asn\n",
      "parsed ('Lys', '127', 'Asn')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Ile120Val\n",
      "parsed ('Ile', '120', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ala289Val\n",
      "parsed ('Ala', '289', 'Val')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Val17Ile\n",
      "parsed ('Val', '17', 'Ile')\n",
      "original Val, amino Val\n",
      "HGVS: p.Lys127Asn\n",
      "parsed ('Lys', '127', 'Asn')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Asp214Gly\n",
      "parsed ('Asp', '214', 'Gly')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Asp123Asn\n",
      "parsed ('Asp', '123', 'Asn')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Leu22Phe\n",
      "parsed ('Leu', '22', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Pro5Ser\n",
      "parsed ('Pro', '5', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Gly306Asp\n",
      "parsed ('Gly', '306', 'Asp')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Ser478Gly\n",
      "parsed ('Ser', '478', 'Gly')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Lys173Arg\n",
      "parsed ('Lys', '173', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Leu364Phe\n",
      "parsed ('Leu', '364', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Thr323Ser\n",
      "parsed ('Thr', '323', 'Ser')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Lys173Glu\n",
      "parsed ('Lys', '173', 'Glu')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Arg372Thr\n",
      "parsed ('Arg', '372', 'Thr')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Pro434Ala\n",
      "parsed ('Pro', '434', 'Ala')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ser188Leu\n",
      "parsed ('Ser', '188', 'Leu')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Arg365Thr\n",
      "parsed ('Arg', '365', 'Thr')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Thr310Arg\n",
      "parsed ('Thr', '310', 'Arg')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Gly167Ala\n",
      "parsed ('Gly', '167', 'Ala')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Gly391Glu\n",
      "parsed ('Gly', '391', 'Glu')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Pro202Ser\n",
      "parsed ('Pro', '202', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Tyr399Cys\n",
      "parsed ('Tyr', '399', 'Cys')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Gly444Asp\n",
      "parsed ('Gly', '444', 'Asp')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Gly385Ala\n",
      "parsed ('Gly', '385', 'Ala')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Leu7Met\n",
      "parsed ('Leu', '7', 'Met')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Gln151His\n",
      "parsed ('Gln', '151', 'His')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Trp12Cys\n",
      "parsed ('Trp', '12', 'Cys')\n",
      "original Trp, amino Trp\n",
      "HGVS: p.Met445Val\n",
      "parsed ('Met', '445', 'Val')\n",
      "original Met, amino Met\n",
      "HGVS: p.Met318Thr\n",
      "parsed ('Met', '318', 'Thr')\n",
      "original Met, amino Met\n",
      "HGVS: p.Lys208Glu\n",
      "parsed ('Lys', '208', 'Glu')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Met358Ile\n",
      "parsed ('Met', '358', 'Ile')\n",
      "original Met, amino Met\n",
      "HGVS: p.Asn384Ser\n",
      "parsed ('Asn', '384', 'Ser')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Thr433Ile\n",
      "parsed ('Thr', '433', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Ile193Val\n",
      "parsed ('Ile', '193', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Arg458Lys\n",
      "parsed ('Arg', '458', 'Lys')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Leu454Pro\n",
      "parsed ('Leu', '454', 'Pro')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Val225Gly\n",
      "parsed ('Val', '225', 'Gly')\n",
      "original Val, amino Val\n",
      "HGVS: p.Pro227Thr\n",
      "parsed ('Pro', '227', 'Thr')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Pro227Ala\n",
      "parsed ('Pro', '227', 'Ala')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Pro231Gln\n",
      "parsed ('Pro', '231', 'Gln')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Leu233Pro\n",
      "parsed ('Leu', '233', 'Pro')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Glu234Gly\n",
      "parsed ('Glu', '234', 'Gly')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Gln265Arg\n",
      "parsed ('Gln', '265', 'Arg')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Leu261Phe\n",
      "parsed ('Leu', '261', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Glu262Asp\n",
      "parsed ('Glu', '262', 'Asp')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Lys143Arg\n",
      "parsed ('Lys', '143', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Phe241Leu\n",
      "parsed ('Phe', '241', 'Leu')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Arg243Thr\n",
      "parsed ('Arg', '243', 'Thr')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Leu142Pro\n",
      "parsed ('Leu', '142', 'Pro')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Ser259Arg\n",
      "parsed ('Ser', '259', 'Arg')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Thr246Ile\n",
      "parsed ('Thr', '246', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Ser259Gly\n",
      "parsed ('Ser', '259', 'Gly')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Asn247His\n",
      "parsed ('Asn', '247', 'His')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Arg255Trp\n",
      "parsed ('Arg', '255', 'Trp')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Arg255Thr\n",
      "parsed ('Arg', '255', 'Thr')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Val253Ala\n",
      "parsed ('Val', '253', 'Ala')\n",
      "original Val, amino Val\n",
      "HGVS: p.Phe248Val\n",
      "parsed ('Phe', '248', 'Val')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Gly73Val\n",
      "parsed ('Gly', '73', 'Val')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Cys58Phe\n",
      "parsed ('Cys', '58', 'Phe')\n",
      "original Cys, amino Cys\n",
      "HGVS: p.Met59Ile\n",
      "parsed ('Met', '59', 'Ile')\n",
      "original Met, amino Met\n",
      "HGVS: p.Met59Val\n",
      "parsed ('Met', '59', 'Val')\n",
      "original Met, amino Met\n",
      "HGVS: p.Gly73Ala\n",
      "parsed ('Gly', '73', 'Ala')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Gly73Asp\n",
      "parsed ('Gly', '73', 'Asp')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Lys55Asn\n",
      "parsed ('Lys', '55', 'Asn')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Gly73Ser\n",
      "parsed ('Gly', '73', 'Ser')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Phe60Ser\n",
      "parsed ('Phe', '60', 'Ser')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.His65Tyr\n",
      "parsed ('His', '65', 'Tyr')\n",
      "original His, amino His\n",
      "HGVS: p.His65Asn\n",
      "parsed ('His', '65', 'Asn')\n",
      "original His, amino His\n",
      "HGVS: p.Asn104Asp\n",
      "parsed ('Asn', '104', 'Asp')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Glu63Lys\n",
      "parsed ('Glu', '63', 'Lys')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Gly26Arg\n",
      "parsed ('Gly', '26', 'Arg')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Cys64Tyr\n",
      "parsed ('Cys', '64', 'Tyr')\n",
      "original Cys, amino Cys\n",
      "HGVS: p.Ser139Arg\n",
      "parsed ('Ser', '139', 'Arg')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Trp72Gly\n",
      "parsed ('Trp', '72', 'Gly')\n",
      "original Trp, amino Trp\n",
      "HGVS: p.Gly26Glu\n",
      "parsed ('Gly', '26', 'Glu')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Glu63Asp\n",
      "parsed ('Glu', '63', 'Asp')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Val101Ala\n",
      "parsed ('Val', '101', 'Ala')\n",
      "original Val, amino Val\n",
      "HGVS: p.Gly69Arg\n",
      "parsed ('Gly', '69', 'Arg')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Lys67Asn\n",
      "parsed ('Lys', '67', 'Asn')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Thr27Ala\n",
      "parsed ('Thr', '27', 'Ala')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Lys70Arg\n",
      "parsed ('Lys', '70', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Lys424Arg\n",
      "parsed ('Lys', '424', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Lys66Gln\n",
      "parsed ('Lys', '66', 'Gln')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Ser100Phe\n",
      "parsed ('Ser', '100', 'Phe')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ser420Ile\n",
      "parsed ('Ser', '420', 'Ile')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Phe74Ile\n",
      "parsed ('Phe', '74', 'Ile')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Glu97Asp\n",
      "parsed ('Glu', '97', 'Asp')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Tyr75Cys\n",
      "parsed ('Tyr', '75', 'Cys')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Gly77Asp\n",
      "parsed ('Gly', '77', 'Asp')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.His28Tyr\n",
      "parsed ('His', '28', 'Tyr')\n",
      "original His, amino His\n",
      "HGVS: p.Lys422Arg\n",
      "parsed ('Lys', '422', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Cys98Arg\n",
      "parsed ('Cys', '98', 'Arg')\n",
      "original Cys, amino Cys\n",
      "HGVS: p.Ser29Pro\n",
      "parsed ('Ser', '29', 'Pro')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Asp425Asn\n",
      "parsed ('Asp', '425', 'Asn')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Leu82Val\n",
      "parsed ('Leu', '82', 'Val')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Thr92Ile\n",
      "parsed ('Thr', '92', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Lys91Gln\n",
      "parsed ('Lys', '91', 'Gln')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Gly48Arg\n",
      "parsed ('Gly', '48', 'Arg')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Val95Gly\n",
      "parsed ('Val', '95', 'Gly')\n",
      "original Val, amino Val\n",
      "HGVS: p.Pro87Ser\n",
      "parsed ('Pro', '87', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ile50Met\n",
      "parsed ('Ile', '50', 'Met')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Val81Ala\n",
      "parsed ('Val', '81', 'Ala')\n",
      "original Val, amino Val\n",
      "HGVS: p.Gly40Glu\n",
      "parsed ('Gly', '40', 'Glu')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Thr85Ile\n",
      "parsed ('Thr', '85', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Val93Gly\n",
      "parsed ('Val', '93', 'Gly')\n",
      "original Val, amino Val\n",
      "HGVS: p.Asn49Ile\n",
      "parsed ('Asn', '49', 'Ile')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Asn49Ser\n",
      "parsed ('Asn', '49', 'Ser')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Ile223Leu\n",
      "parsed ('Ile', '223', 'Leu')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Met89Ile\n",
      "parsed ('Met', '89', 'Ile')\n",
      "original Met, amino Met\n",
      "HGVS: p.Ile84Thr\n",
      "parsed ('Ile', '84', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Leu32Val\n",
      "parsed ('Leu', '32', 'Val')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Ile90Val\n",
      "parsed ('Ile', '90', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Met114Ile\n",
      "parsed ('Met', '114', 'Ile')\n",
      "original Met, amino Met\n",
      "HGVS: p.Ile223Met\n",
      "parsed ('Ile', '223', 'Met')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Arg130Gly\n",
      "parsed ('Arg', '130', 'Gly')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Lys34Glu\n",
      "parsed ('Lys', '34', 'Glu')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Arg130Pro\n",
      "parsed ('Arg', '130', 'Pro')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.His267Asn\n",
      "parsed ('His', '267', 'Asn')\n",
      "original His, amino His\n",
      "HGVS: p.Pro41His\n",
      "parsed ('Pro', '41', 'His')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Thr42Ala\n",
      "parsed ('Thr', '42', 'Ala')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Lys282Glu\n",
      "parsed ('Lys', '282', 'Glu')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Thr42Pro\n",
      "parsed ('Thr', '42', 'Pro')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Ala117Val\n",
      "parsed ('Ala', '117', 'Val')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.His267Leu\n",
      "parsed ('His', '267', 'Leu')\n",
      "original His, amino His\n",
      "HGVS: p.Leu36Val\n",
      "parsed ('Leu', '36', 'Val')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.His287Tyr\n",
      "parsed ('His', '287', 'Tyr')\n",
      "original His, amino His\n",
      "HGVS: p.Lys35Arg\n",
      "parsed ('Lys', '35', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Pro474Ser\n",
      "parsed ('Pro', '474', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ser286Pro\n",
      "parsed ('Ser', '286', 'Pro')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Pro43Arg\n",
      "parsed ('Pro', '43', 'Arg')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Pro41Ser\n",
      "parsed ('Pro', '41', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ala2Gly\n",
      "parsed ('Ala', '2', 'Gly')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Gln273Arg\n",
      "parsed ('Gln', '273', 'Arg')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Glu122Gln\n",
      "parsed ('Glu', '122', 'Gln')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Ile427Thr\n",
      "parsed ('Ile', '427', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Leu36Arg\n",
      "parsed ('Leu', '36', 'Arg')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Ser478Arg\n",
      "parsed ('Ser', '478', 'Arg')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Val296Leu\n",
      "parsed ('Val', '296', 'Leu')\n",
      "original Val, amino Val\n",
      "HGVS: p.Asp154Gly\n",
      "parsed ('Asp', '154', 'Gly')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Glu285Val\n",
      "parsed ('Glu', '285', 'Val')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Ala121Asp\n",
      "parsed ('Ala', '121', 'Asp')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Ala121Gly\n",
      "parsed ('Ala', '121', 'Gly')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Pro429Arg\n",
      "parsed ('Pro', '429', 'Arg')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Met181Ile\n",
      "parsed ('Met', '181', 'Ile')\n",
      "original Met, amino Met\n",
      "HGVS: p.Pro218Thr\n",
      "parsed ('Pro', '218', 'Thr')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Asp182Asn\n",
      "parsed ('Asp', '182', 'Asn')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Ser119Thr\n",
      "parsed ('Ser', '119', 'Thr')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ser501Gly\n",
      "parsed ('Ser', '501', 'Gly')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ser501Cys\n",
      "parsed ('Ser', '501', 'Cys')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Leu3Phe\n",
      "parsed ('Leu', '3', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Ser501Asn\n",
      "parsed ('Ser', '501', 'Asn')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ala337Val\n",
      "parsed ('Ala', '337', 'Val')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Arg128Lys\n",
      "parsed ('Arg', '128', 'Lys')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Glu122Lys\n",
      "parsed ('Glu', '122', 'Lys')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Ser18Arg\n",
      "parsed ('Ser', '18', 'Arg')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Leu19Gln\n",
      "parsed ('Leu', '19', 'Gln')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Asp123Val\n",
      "parsed ('Asp', '123', 'Val')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Thr309Ile\n",
      "parsed ('Thr', '309', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Ser18Asn\n",
      "parsed ('Ser', '18', 'Asn')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Cys468Trp\n",
      "parsed ('Cys', '468', 'Trp')\n",
      "original Cys, amino Cys\n",
      "HGVS: p.Phe176Leu\n",
      "parsed ('Phe', '176', 'Leu')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Glu470Gln\n",
      "parsed ('Glu', '470', 'Gln')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Arg162Gly\n",
      "parsed ('Arg', '162', 'Gly')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Asp217Val\n",
      "parsed ('Asp', '217', 'Val')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Val394Ala\n",
      "parsed ('Val', '394', 'Ala')\n",
      "original Val, amino Val\n",
      "HGVS: p.Gly306Ala\n",
      "parsed ('Gly', '306', 'Ala')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Val191Ala\n",
      "parsed ('Val', '191', 'Ala')\n",
      "original Val, amino Val\n",
      "HGVS: p.Glu294Gly\n",
      "parsed ('Glu', '294', 'Gly')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Asp336Val\n",
      "parsed ('Asp', '336', 'Val')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Val500Leu\n",
      "parsed ('Val', '500', 'Leu')\n",
      "original Val, amino Val\n",
      "HGVS: p.Gly385Glu\n",
      "parsed ('Gly', '385', 'Glu')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.His324Gln\n",
      "parsed ('His', '324', 'Gln')\n",
      "original His, amino His\n",
      "HGVS: p.Ile303Thr\n",
      "parsed ('Ile', '303', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Pro340Ser\n",
      "parsed ('Pro', '340', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Pro340Thr\n",
      "parsed ('Pro', '340', 'Thr')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Pro397Leu\n",
      "parsed ('Pro', '397', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Arg375Met\n",
      "parsed ('Arg', '375', 'Met')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Glu486Gly\n",
      "parsed ('Glu', '486', 'Gly')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Lys487Gln\n",
      "parsed ('Lys', '487', 'Gln')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Pro169Ser\n",
      "parsed ('Pro', '169', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ser291Phe\n",
      "parsed ('Ser', '291', 'Phe')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Lys342Asn\n",
      "parsed ('Lys', '342', 'Asn')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Leu491Pro\n",
      "parsed ('Leu', '491', 'Pro')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Arg372Ile\n",
      "parsed ('Arg', '372', 'Ile')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Ala337Gly\n",
      "parsed ('Ala', '337', 'Gly')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Lys208Arg\n",
      "parsed ('Lys', '208', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Ala337Glu\n",
      "parsed ('Ala', '337', 'Glu')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Ser180Gly\n",
      "parsed ('Ser', '180', 'Gly')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ala343Val\n",
      "parsed ('Ala', '343', 'Val')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Tyr319Cys\n",
      "parsed ('Tyr', '319', 'Cys')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Asn206Tyr\n",
      "parsed ('Asn', '206', 'Tyr')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Met318Val\n",
      "parsed ('Met', '318', 'Val')\n",
      "original Met, amino Met\n",
      "HGVS: p.Leu22Arg\n",
      "parsed ('Leu', '22', 'Arg')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Val157Ala\n",
      "parsed ('Val', '157', 'Ala')\n",
      "original Val, amino Val\n",
      "HGVS: p.Lys379Arg\n",
      "parsed ('Lys', '379', 'Arg')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Met386Ile\n",
      "parsed ('Met', '386', 'Ile')\n",
      "original Met, amino Met\n",
      "HGVS: p.Ile184Met\n",
      "parsed ('Ile', '184', 'Met')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ile396Met\n",
      "parsed ('Ile', '396', 'Met')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Asn159Ser\n",
      "parsed ('Asn', '159', 'Ser')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Glu165Asp\n",
      "parsed ('Glu', '165', 'Asp')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Leu14Phe\n",
      "parsed ('Leu', '14', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Met181Thr\n",
      "parsed ('Met', '181', 'Thr')\n",
      "original Met, amino Met\n",
      "HGVS: p.Tyr355His\n",
      "parsed ('Tyr', '355', 'His')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Pro405Thr\n",
      "parsed ('Pro', '405', 'Thr')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ile396Thr\n",
      "parsed ('Ile', '396', 'Thr')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ile301Met\n",
      "parsed ('Ile', '301', 'Met')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Pro488Thr\n",
      "parsed ('Pro', '488', 'Thr')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Pro488Ser\n",
      "parsed ('Pro', '488', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Gly391Val\n",
      "parsed ('Gly', '391', 'Val')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Pro202Thr\n",
      "parsed ('Pro', '202', 'Thr')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Ser312Arg\n",
      "parsed ('Ser', '312', 'Arg')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Pro345Thr\n",
      "parsed ('Pro', '345', 'Thr')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Thr166Ile\n",
      "parsed ('Thr', '166', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Ser312Gly\n",
      "parsed ('Ser', '312', 'Gly')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Thr207Ile\n",
      "parsed ('Thr', '207', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Tyr399Ser\n",
      "parsed ('Tyr', '399', 'Ser')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Gly167Arg\n",
      "parsed ('Gly', '167', 'Arg')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Asp194Tyr\n",
      "parsed ('Asp', '194', 'Tyr')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Leu482His\n",
      "parsed ('Leu', '482', 'His')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Arg458Ile\n",
      "parsed ('Arg', '458', 'Ile')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Val394Leu\n",
      "parsed ('Val', '394', 'Leu')\n",
      "original Val, amino Val\n",
      "HGVS: p.Ser398Gly\n",
      "parsed ('Ser', '398', 'Gly')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Ser398Cys\n",
      "parsed ('Ser', '398', 'Cys')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Asp497Tyr\n",
      "parsed ('Asp', '497', 'Tyr')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Asp6Glu\n",
      "parsed ('Asp', '6', 'Glu')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Ala8Val\n",
      "parsed ('Ala', '8', 'Val')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Ser398Arg\n",
      "parsed ('Ser', '398', 'Arg')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Val146Phe\n",
      "parsed ('Val', '146', 'Phe')\n",
      "original Val, amino Val\n",
      "HGVS: p.Glu333Gln\n",
      "parsed ('Glu', '333', 'Gln')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Tyr430His\n",
      "parsed ('Tyr', '430', 'His')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Met353Ile\n",
      "parsed ('Met', '353', 'Ile')\n",
      "original Met, amino Met\n",
      "HGVS: p.Gly385Val\n",
      "parsed ('Gly', '385', 'Val')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Thr323Ser\n",
      "parsed ('Thr', '323', 'Ser')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Leu314Pro\n",
      "parsed ('Leu', '314', 'Pro')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Pro411Ala\n",
      "parsed ('Pro', '411', 'Ala')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Val350Met\n",
      "parsed ('Val', '350', 'Met')\n",
      "original Val, amino Val\n",
      "HGVS: p.Gln484Arg\n",
      "parsed ('Gln', '484', 'Arg')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Val392Met\n",
      "parsed ('Val', '392', 'Met')\n",
      "original Val, amino Val\n",
      "HGVS: p.Asp201Asn\n",
      "parsed ('Asp', '201', 'Asn')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Lys487Glu\n",
      "parsed ('Lys', '487', 'Glu')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Arg418Lys\n",
      "parsed ('Arg', '418', 'Lys')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.His402Tyr\n",
      "parsed ('His', '402', 'Tyr')\n",
      "original His, amino His\n",
      "HGVS: p.Gly391Arg\n",
      "parsed ('Gly', '391', 'Arg')\n",
      "original Gly, amino Gly\n",
      "HGVS: p.Lys208Gln\n",
      "parsed ('Lys', '208', 'Gln')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Met318Arg\n",
      "parsed ('Met', '318', 'Arg')\n",
      "original Met, amino Met\n",
      "HGVS: p.Glu354Gly\n",
      "parsed ('Glu', '354', 'Gly')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Pro199Ser\n",
      "parsed ('Pro', '199', 'Ser')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Glu333Gly\n",
      "parsed ('Glu', '333', 'Gly')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Lys390Glu\n",
      "parsed ('Lys', '390', 'Glu')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Pro199Leu\n",
      "parsed ('Pro', '199', 'Leu')\n",
      "original Pro, amino Pro\n",
      "HGVS: p.Thr349Ala\n",
      "parsed ('Thr', '349', 'Ala')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Glu10Gln\n",
      "parsed ('Glu', '10', 'Gln')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Gln332Arg\n",
      "parsed ('Gln', '332', 'Arg')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Ile388Phe\n",
      "parsed ('Ile', '388', 'Phe')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Ser195Cys\n",
      "parsed ('Ser', '195', 'Cys')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Leu210Pro\n",
      "parsed ('Leu', '210', 'Pro')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Leu13Phe\n",
      "parsed ('Leu', '13', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Glu205Asp\n",
      "parsed ('Glu', '205', 'Asp')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Thr349Ile\n",
      "parsed ('Thr', '349', 'Ile')\n",
      "original Thr, amino Thr\n",
      "HGVS: p.Tyr432His\n",
      "parsed ('Tyr', '432', 'His')\n",
      "original Tyr, amino Tyr\n",
      "HGVS: p.Glu374Asp\n",
      "parsed ('Glu', '374', 'Asp')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Arg365Ile\n",
      "parsed ('Arg', '365', 'Ile')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Glu374Lys\n",
      "parsed ('Glu', '374', 'Lys')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Asp348Glu\n",
      "parsed ('Asp', '348', 'Glu')\n",
      "original Asp, amino Asp\n",
      "HGVS: p.Lys413Asn\n",
      "parsed ('Lys', '413', 'Asn')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Asn361Asp\n",
      "parsed ('Asn', '361', 'Asp')\n",
      "original Asn, amino Asn\n",
      "HGVS: p.Lys406Glu\n",
      "parsed ('Lys', '406', 'Glu')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Glu382Ala\n",
      "parsed ('Glu', '382', 'Ala')\n",
      "original Glu, amino Glu\n",
      "HGVS: p.Arg446Ser\n",
      "parsed ('Arg', '446', 'Ser')\n",
      "original Arg, amino Arg\n",
      "HGVS: p.Ala16Gly\n",
      "parsed ('Ala', '16', 'Gly')\n",
      "original Ala, amino Ala\n",
      "HGVS: p.Ile383Asn\n",
      "parsed ('Ile', '383', 'Asn')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Trp12Arg\n",
      "parsed ('Trp', '12', 'Arg')\n",
      "original Trp, amino Trp\n",
      "HGVS: p.Ile431Val\n",
      "parsed ('Ile', '431', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Leu460Phe\n",
      "parsed ('Leu', '460', 'Phe')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Ser195Ala\n",
      "parsed ('Ser', '195', 'Ala')\n",
      "original Ser, amino Ser\n",
      "HGVS: p.Phe447Val\n",
      "parsed ('Phe', '447', 'Val')\n",
      "original Phe, amino Phe\n",
      "HGVS: p.Lys378Gln\n",
      "parsed ('Lys', '378', 'Gln')\n",
      "original Lys, amino Lys\n",
      "HGVS: p.Met371Thr\n",
      "parsed ('Met', '371', 'Thr')\n",
      "original Met, amino Met\n",
      "HGVS: p.Gln200Arg\n",
      "parsed ('Gln', '200', 'Arg')\n",
      "original Gln, amino Gln\n",
      "HGVS: p.Ile443Val\n",
      "parsed ('Ile', '443', 'Val')\n",
      "original Ile, amino Ile\n",
      "HGVS: p.Leu449Pro\n",
      "parsed ('Leu', '449', 'Pro')\n",
      "original Leu, amino Leu\n",
      "HGVS: p.Met452Val\n",
      "parsed ('Met', '452', 'Val')\n",
      "original Met, amino Met\n",
      "            gnomAD ID  Chromosome  Position         rsIDs Reference Alternate  \\\n",
      "0      7-99758061-A-G           7  99758061           NaN         A         G   \n",
      "1      7-99758062-A-G           7  99758062           NaN         A         G   \n",
      "2      7-99758063-G-A           7  99758063  rs1815222939         G         A   \n",
      "3      7-99758065-A-G           7  99758065           NaN         A         G   \n",
      "4      7-99758066-A-G           7  99758066           NaN         A         G   \n",
      "...               ...         ...       ...           ...       ...       ...   \n",
      "2082   7-99784151-A-G           7  99784151  rs1208976807         A         G   \n",
      "2083   7-99784151-A-T           7  99784151           NaN         A         T   \n",
      "2084  7-99784151-AT-A           7  99784151           NaN        AT         A   \n",
      "2085   7-99784152-T-C           7  99784152           NaN         T         C   \n",
      "2086   7-99784155-G-A           7  99784155  rs1438109235         G         A   \n",
      "\n",
      "                            Source Filters - exomes Filters - genomes  \\\n",
      "0                    gnomAD Exomes             PASS               NaN   \n",
      "1                    gnomAD Exomes             PASS               NaN   \n",
      "2                    gnomAD Exomes             PASS               NaN   \n",
      "3                    gnomAD Exomes             PASS               NaN   \n",
      "4                    gnomAD Exomes             PASS               NaN   \n",
      "...                            ...              ...               ...   \n",
      "2082  gnomAD Exomes,gnomAD Genomes             PASS              PASS   \n",
      "2083                 gnomAD Exomes             PASS               NaN   \n",
      "2084                 gnomAD Exomes             PASS               NaN   \n",
      "2085                 gnomAD Exomes             PASS               NaN   \n",
      "2086  gnomAD Exomes,gnomAD Genomes             PASS              PASS   \n",
      "\n",
      "             Transcript  ... Homozygote Count Amish Hemizygote Count Amish  \\\n",
      "0     ENST00000651514.1  ...                      0                      0   \n",
      "1     ENST00000651514.1  ...                      0                      0   \n",
      "2     ENST00000651514.1  ...                      0                      0   \n",
      "3     ENST00000651514.1  ...                      0                      0   \n",
      "4     ENST00000651514.1  ...                      0                      0   \n",
      "...                 ...  ...                    ...                    ...   \n",
      "2082  ENST00000651514.1  ...                      0                      0   \n",
      "2083  ENST00000651514.1  ...                      0                      0   \n",
      "2084  ENST00000651514.1  ...                      0                      0   \n",
      "2085  ENST00000651514.1  ...                      0                      0   \n",
      "2086  ENST00000651514.1  ...                      0                      0   \n",
      "\n",
      "     Allele Count South Asian Allele Number South Asian  \\\n",
      "0                           0                     82904   \n",
      "1                           0                     83082   \n",
      "2                           1                     83300   \n",
      "3                           0                     83678   \n",
      "4                           0                     84100   \n",
      "...                       ...                       ...   \n",
      "2082                        0                     86926   \n",
      "2083                        0                     86922   \n",
      "2084                        0                     86922   \n",
      "2085                        0                     86952   \n",
      "2086                        3                     86148   \n",
      "\n",
      "     Homozygote Count South Asian  Hemizygote Count South Asian  \\\n",
      "0                               0                             0   \n",
      "1                               0                             0   \n",
      "2                               0                             0   \n",
      "3                               0                             0   \n",
      "4                               0                             0   \n",
      "...                           ...                           ...   \n",
      "2082                            0                             0   \n",
      "2083                            0                             0   \n",
      "2084                            0                             0   \n",
      "2085                            0                             0   \n",
      "2086                            0                             0   \n",
      "\n",
      "     Allele Count Remaining  Allele Number Remaining  \\\n",
      "0                         0                    49870   \n",
      "1                         0                    50164   \n",
      "2                         0                    50392   \n",
      "3                         0                    50896   \n",
      "4                         0                    51460   \n",
      "...                     ...                      ...   \n",
      "2082                      2                    56174   \n",
      "2083                      0                    56196   \n",
      "2084                      0                    56196   \n",
      "2085                      0                    56234   \n",
      "2086                      0                    55032   \n",
      "\n",
      "      Homozygote Count Remaining  Hemizygote Count Remaining  \n",
      "0                              0                           0  \n",
      "1                              0                           0  \n",
      "2                              0                           0  \n",
      "3                              0                           0  \n",
      "4                              0                           0  \n",
      "...                          ...                         ...  \n",
      "2082                           0                           0  \n",
      "2083                           0                           0  \n",
      "2084                           0                           0  \n",
      "2085                           0                           0  \n",
      "2086                           0                           0  \n",
      "\n",
      "[2087 rows x 72 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_120446/2873550858.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  relevant_veps[\"mutant_seq\"] = relevant_veps[\"Protein Consequence\"].apply(lambda x: apply_hgvs_annotation(ref_seq, x))\n"
     ]
    }
   ],
   "source": [
    "process_data(\"CYP3A4\", data[\"CYP3A4\"][0], data[\"CYP3A4\"][1], mis_VEPs, canonical_MANE_transcripts[\"CYP3A4\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "a04e906d-99cb-4d8d-8aa4-c3cfae774eb5",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"CYP3A4/CYP3A4_gnomAD_v4.1.0_ENSG00000160868.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "19a5b1ad-95f5-4670-a37c-ec03ecf1e915",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "VEP Annotation\n",
       "intron_variant             1055\n",
       "missense_variant            586\n",
       "synonymous_variant          202\n",
       "splice_region_variant        63\n",
       "5_prime_UTR_variant          45\n",
       "3_prime_UTR_variant          40\n",
       "frameshift_variant           35\n",
       "stop_gained                  24\n",
       "splice_donor_variant         16\n",
       "splice_acceptor_variant       9\n",
       "inframe_deletion              8\n",
       "stop_lost                     2\n",
       "stop_retained_variant         1\n",
       "inframe_insertion             1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"VEP Annotation\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "e7085e78-bda1-4a0c-9ba0-c0c7fb16b1d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gnomAD ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>rsIDs</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternate</th>\n",
       "      <th>Source</th>\n",
       "      <th>Filters - exomes</th>\n",
       "      <th>Filters - genomes</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>...</th>\n",
       "      <th>Homozygote Count Amish</th>\n",
       "      <th>Hemizygote Count Amish</th>\n",
       "      <th>Allele Count South Asian</th>\n",
       "      <th>Allele Number South Asian</th>\n",
       "      <th>Homozygote Count South Asian</th>\n",
       "      <th>Hemizygote Count South Asian</th>\n",
       "      <th>Allele Count Remaining</th>\n",
       "      <th>Allele Number Remaining</th>\n",
       "      <th>Homozygote Count Remaining</th>\n",
       "      <th>Hemizygote Count Remaining</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7-99758061-A-G</td>\n",
       "      <td>7</td>\n",
       "      <td>99758061</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>82904</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>49870</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7-99758062-A-G</td>\n",
       "      <td>7</td>\n",
       "      <td>99758062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>83082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>50164</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7-99758063-G-A</td>\n",
       "      <td>7</td>\n",
       "      <td>99758063</td>\n",
       "      <td>rs1815222939</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>83300</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>50392</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7-99758065-A-G</td>\n",
       "      <td>7</td>\n",
       "      <td>99758065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>83678</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>50896</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7-99758066-A-G</td>\n",
       "      <td>7</td>\n",
       "      <td>99758066</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>84100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>51460</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  72 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gnomAD ID  Chromosome  Position         rsIDs Reference Alternate  \\\n",
       "0  7-99758061-A-G           7  99758061           NaN         A         G   \n",
       "1  7-99758062-A-G           7  99758062           NaN         A         G   \n",
       "2  7-99758063-G-A           7  99758063  rs1815222939         G         A   \n",
       "3  7-99758065-A-G           7  99758065           NaN         A         G   \n",
       "4  7-99758066-A-G           7  99758066           NaN         A         G   \n",
       "\n",
       "          Source Filters - exomes Filters - genomes         Transcript  ...  \\\n",
       "0  gnomAD Exomes             PASS               NaN  ENST00000651514.1  ...   \n",
       "1  gnomAD Exomes             PASS               NaN  ENST00000651514.1  ...   \n",
       "2  gnomAD Exomes             PASS               NaN  ENST00000651514.1  ...   \n",
       "3  gnomAD Exomes             PASS               NaN  ENST00000651514.1  ...   \n",
       "4  gnomAD Exomes             PASS               NaN  ENST00000651514.1  ...   \n",
       "\n",
       "  Homozygote Count Amish Hemizygote Count Amish Allele Count South Asian  \\\n",
       "0                      0                      0                        0   \n",
       "1                      0                      0                        0   \n",
       "2                      0                      0                        1   \n",
       "3                      0                      0                        0   \n",
       "4                      0                      0                        0   \n",
       "\n",
       "  Allele Number South Asian Homozygote Count South Asian  \\\n",
       "0                     82904                            0   \n",
       "1                     83082                            0   \n",
       "2                     83300                            0   \n",
       "3                     83678                            0   \n",
       "4                     84100                            0   \n",
       "\n",
       "   Hemizygote Count South Asian Allele Count Remaining  \\\n",
       "0                             0                      0   \n",
       "1                             0                      0   \n",
       "2                             0                      0   \n",
       "3                             0                      0   \n",
       "4                             0                      0   \n",
       "\n",
       "   Allele Number Remaining  Homozygote Count Remaining  \\\n",
       "0                    49870                           0   \n",
       "1                    50164                           0   \n",
       "2                    50392                           0   \n",
       "3                    50896                           0   \n",
       "4                    51460                           0   \n",
       "\n",
       "   Hemizygote Count Remaining  \n",
       "0                           0  \n",
       "1                           0  \n",
       "2                           0  \n",
       "3                           0  \n",
       "4                           0  \n",
       "\n",
       "[5 rows x 72 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "495f742a-3c57-49b0-94d5-c891d4d35945",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1057    9.350962e-01\n",
       "632     1.532552e-01\n",
       "1212    1.492865e-02\n",
       "276     1.473407e-02\n",
       "229     6.228435e-03\n",
       "            ...     \n",
       "212     6.195265e-07\n",
       "1109    6.195257e-07\n",
       "234     6.195219e-07\n",
       "223     6.195088e-07\n",
       "218     6.195073e-07\n",
       "Name: Allele Frequency, Length: 2087, dtype: float64"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.sort_values(\"Allele Frequency\", ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "33404477-8604-4c75-a03f-a19275d45f64",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       2.424803e-06\n",
       "1       8.034852e-07\n",
       "2       7.984032e-07\n",
       "3       7.894953e-07\n",
       "4       1.555556e-06\n",
       "            ...     \n",
       "2082    4.908340e-06\n",
       "2083    7.011335e-07\n",
       "2084    7.011335e-07\n",
       "2085    1.400143e-06\n",
       "2086    2.869412e-06\n",
       "Name: Allele Frequency, Length: 2087, dtype: float64"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"Allele Frequency\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "bca0145e-7e70-4eba-9a1d-d5c5a1bbf321",
   "metadata": {},
   "outputs": [],
   "source": [
    "relevant_veps = df[df[\"VEP Annotation\"].isin(relevant_VEPs)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "ee2c64db-9476-46af-8c7a-bdbf304a9b3c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gnomAD ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>rsIDs</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternate</th>\n",
       "      <th>Source</th>\n",
       "      <th>Filters - exomes</th>\n",
       "      <th>Filters - genomes</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>...</th>\n",
       "      <th>Homozygote Count Amish</th>\n",
       "      <th>Hemizygote Count Amish</th>\n",
       "      <th>Allele Count South Asian</th>\n",
       "      <th>Allele Number South Asian</th>\n",
       "      <th>Homozygote Count South Asian</th>\n",
       "      <th>Hemizygote Count South Asian</th>\n",
       "      <th>Allele Count Remaining</th>\n",
       "      <th>Allele Number Remaining</th>\n",
       "      <th>Homozygote Count Remaining</th>\n",
       "      <th>Hemizygote Count Remaining</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>7-99758133-T-G</td>\n",
       "      <td>7</td>\n",
       "      <td>99758133</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91076</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62476</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>7-99758133-T-C</td>\n",
       "      <td>7</td>\n",
       "      <td>99758133</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91076</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>62476</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>7-99758140-C-T</td>\n",
       "      <td>7</td>\n",
       "      <td>99758140</td>\n",
       "      <td>rs775072566</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>gnomAD Exomes,gnomAD Genomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91088</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62472</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>7-99758143-C-T</td>\n",
       "      <td>7</td>\n",
       "      <td>99758143</td>\n",
       "      <td>rs1286121238</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>gnomAD Genomes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91092</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62476</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>7-99758144-T-A</td>\n",
       "      <td>7</td>\n",
       "      <td>99758144</td>\n",
       "      <td>rs1221668126</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>gnomAD Genomes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91092</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62472</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2037</th>\n",
       "      <td>7-99784075-G-A</td>\n",
       "      <td>7</td>\n",
       "      <td>99784075</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62474</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2038</th>\n",
       "      <td>7-99784077-G-T</td>\n",
       "      <td>7</td>\n",
       "      <td>99784077</td>\n",
       "      <td>rs371360704</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>gnomAD Exomes,gnomAD Genomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>91072</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62478</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2039</th>\n",
       "      <td>7-99784077-G-C</td>\n",
       "      <td>7</td>\n",
       "      <td>99784077</td>\n",
       "      <td>rs371360704</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91072</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62478</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2040</th>\n",
       "      <td>7-99784077-G-A</td>\n",
       "      <td>7</td>\n",
       "      <td>99784077</td>\n",
       "      <td>rs371360704</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91072</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>62478</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2041</th>\n",
       "      <td>7-99784078-C-T</td>\n",
       "      <td>7</td>\n",
       "      <td>99784078</td>\n",
       "      <td>rs1226205448</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>gnomAD Exomes,gnomAD Genomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ENST00000651514.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91072</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>62452</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>655 rows  72 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           gnomAD ID  Chromosome  Position         rsIDs Reference Alternate  \\\n",
       "40    7-99758133-T-G           7  99758133           NaN         T         G   \n",
       "41    7-99758133-T-C           7  99758133           NaN         T         C   \n",
       "45    7-99758140-C-T           7  99758140   rs775072566         C         T   \n",
       "47    7-99758143-C-T           7  99758143  rs1286121238         C         T   \n",
       "48    7-99758144-T-A           7  99758144  rs1221668126         T         A   \n",
       "...              ...         ...       ...           ...       ...       ...   \n",
       "2037  7-99784075-G-A           7  99784075           NaN         G         A   \n",
       "2038  7-99784077-G-T           7  99784077   rs371360704         G         T   \n",
       "2039  7-99784077-G-C           7  99784077   rs371360704         G         C   \n",
       "2040  7-99784077-G-A           7  99784077   rs371360704         G         A   \n",
       "2041  7-99784078-C-T           7  99784078  rs1226205448         C         T   \n",
       "\n",
       "                            Source Filters - exomes Filters - genomes  \\\n",
       "40                   gnomAD Exomes             PASS               NaN   \n",
       "41                   gnomAD Exomes             PASS               NaN   \n",
       "45    gnomAD Exomes,gnomAD Genomes             PASS              PASS   \n",
       "47                  gnomAD Genomes              NaN              PASS   \n",
       "48                  gnomAD Genomes              NaN              PASS   \n",
       "...                            ...              ...               ...   \n",
       "2037                 gnomAD Exomes             PASS               NaN   \n",
       "2038  gnomAD Exomes,gnomAD Genomes             PASS              PASS   \n",
       "2039                 gnomAD Exomes             PASS               NaN   \n",
       "2040                 gnomAD Exomes             PASS               NaN   \n",
       "2041  gnomAD Exomes,gnomAD Genomes             PASS              PASS   \n",
       "\n",
       "             Transcript  ... Homozygote Count Amish Hemizygote Count Amish  \\\n",
       "40    ENST00000651514.1  ...                      0                      0   \n",
       "41    ENST00000651514.1  ...                      0                      0   \n",
       "45    ENST00000651514.1  ...                      0                      0   \n",
       "47    ENST00000651514.1  ...                      0                      0   \n",
       "48    ENST00000651514.1  ...                      0                      0   \n",
       "...                 ...  ...                    ...                    ...   \n",
       "2037  ENST00000651514.1  ...                      0                      0   \n",
       "2038  ENST00000651514.1  ...                      0                      0   \n",
       "2039  ENST00000651514.1  ...                      0                      0   \n",
       "2040  ENST00000651514.1  ...                      0                      0   \n",
       "2041  ENST00000651514.1  ...                      0                      0   \n",
       "\n",
       "     Allele Count South Asian Allele Number South Asian  \\\n",
       "40                          0                     91076   \n",
       "41                          0                     91076   \n",
       "45                          0                     91088   \n",
       "47                          0                     91092   \n",
       "48                          0                     91092   \n",
       "...                       ...                       ...   \n",
       "2037                        0                     91082   \n",
       "2038                        1                     91072   \n",
       "2039                        0                     91072   \n",
       "2040                        0                     91072   \n",
       "2041                        0                     91072   \n",
       "\n",
       "     Homozygote Count South Asian  Hemizygote Count South Asian  \\\n",
       "40                              0                             0   \n",
       "41                              0                             0   \n",
       "45                              0                             0   \n",
       "47                              0                             0   \n",
       "48                              0                             0   \n",
       "...                           ...                           ...   \n",
       "2037                            0                             0   \n",
       "2038                            0                             0   \n",
       "2039                            0                             0   \n",
       "2040                            0                             0   \n",
       "2041                            0                             0   \n",
       "\n",
       "     Allele Count Remaining  Allele Number Remaining  \\\n",
       "40                        0                    62476   \n",
       "41                        1                    62476   \n",
       "45                        0                    62472   \n",
       "47                        0                    62476   \n",
       "48                        0                    62472   \n",
       "...                     ...                      ...   \n",
       "2037                      0                    62474   \n",
       "2038                      0                    62478   \n",
       "2039                      0                    62478   \n",
       "2040                      1                    62478   \n",
       "2041                      2                    62452   \n",
       "\n",
       "      Homozygote Count Remaining  Hemizygote Count Remaining  \n",
       "40                             0                           0  \n",
       "41                             0                           0  \n",
       "45                             0                           0  \n",
       "47                             0                           0  \n",
       "48                             0                           0  \n",
       "...                          ...                         ...  \n",
       "2037                           0                           0  \n",
       "2038                           0                           0  \n",
       "2039                           0                           0  \n",
       "2040                           0                           0  \n",
       "2041                           0                           0  \n",
       "\n",
       "[655 rows x 72 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relevant_veps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "7b845655-231e-42c8-8b2b-ca7bf715c8ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "VEP Annotation\n",
       "missense_variant      586\n",
       "frameshift_variant     35\n",
       "stop_gained            24\n",
       "inframe_deletion        8\n",
       "stop_lost               2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "5115e73f-632e-49b5-a266-2cbf0b5eb720",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein Consequence\n",
       "p.Lys421Arg          2\n",
       "p.Thr323Ser          2\n",
       "p.Trp12GlyfsTer78    2\n",
       "p.Met114Ile          2\n",
       "p.Glu125Asp          2\n",
       "                    ..\n",
       "p.Ile335Thr          1\n",
       "p.Glu334Ter          1\n",
       "p.Glu333Gly          1\n",
       "p.Glu333Gln          1\n",
       "p.Pro340Ser          1\n",
       "Name: count, Length: 645, dtype: int64"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relevant_veps['Protein Consequence'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f2409342-b5da-44b9-967e-2f16179beb95",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "VEP Annotation\n",
       "intron_variant             1055\n",
       "missense_variant            586\n",
       "synonymous_variant          202\n",
       "splice_region_variant        63\n",
       "5_prime_UTR_variant          45\n",
       "3_prime_UTR_variant          40\n",
       "frameshift_variant           35\n",
       "stop_gained                  24\n",
       "splice_donor_variant         16\n",
       "splice_acceptor_variant       9\n",
       "inframe_deletion              8\n",
       "stop_lost                     2\n",
       "stop_retained_variant         1\n",
       "inframe_insertion             1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['VEP Annotation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "6848c687-0651-4ea7-9d24-3c17eed614e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "196                  p.Ser464del\n",
       "263                  p.Lys422del\n",
       "461                  p.Val394del\n",
       "523                  p.Val360del\n",
       "668                  p.Ala322del\n",
       "672     p.Met318_Glu320delinsLys\n",
       "1089                 p.Lys209del\n",
       "2013                  p.Leu15del\n",
       "Name: Protein Consequence, dtype: object"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['VEP Annotation'] == 'inframe_deletion'][\"Protein Consequence\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5669d25d-fe62-4bcd-985c-de9dbef6e971",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gnomAD ID', 'Chromosome', 'Position', 'rsIDs', 'Reference',\n",
       "       'Alternate', 'Source', 'Filters - exomes', 'Filters - genomes',\n",
       "       'Transcript', 'HGVS Consequence', 'Protein Consequence',\n",
       "       'Transcript Consequence', 'VEP Annotation',\n",
       "       'ClinVar Germline Classification', 'ClinVar Variation ID', 'Flags',\n",
       "       'Allele Count', 'Allele Number', 'Allele Frequency', 'Homozygote Count',\n",
       "       'Hemizygote Count', 'Filters - joint', 'GroupMax FAF group',\n",
       "       'GroupMax FAF frequency', 'cadd', 'revel_max', 'spliceai_ds_max',\n",
       "       'pangolin_largest_ds', 'phylop', 'sift_max', 'polyphen_max',\n",
       "       'Allele Count African/African American',\n",
       "       'Allele Number African/African American',\n",
       "       'Homozygote Count African/African American',\n",
       "       'Hemizygote Count African/African American',\n",
       "       'Allele Count Admixed American', 'Allele Number Admixed American',\n",
       "       'Homozygote Count Admixed American',\n",
       "       'Hemizygote Count Admixed American', 'Allele Count Ashkenazi Jewish',\n",
       "       'Allele Number Ashkenazi Jewish', 'Homozygote Count Ashkenazi Jewish',\n",
       "       'Hemizygote Count Ashkenazi Jewish', 'Allele Count East Asian',\n",
       "       'Allele Number East Asian', 'Homozygote Count East Asian',\n",
       "       'Hemizygote Count East Asian', 'Allele Count European (Finnish)',\n",
       "       'Allele Number European (Finnish)',\n",
       "       'Homozygote Count European (Finnish)',\n",
       "       'Hemizygote Count European (Finnish)', 'Allele Count Middle Eastern',\n",
       "       'Allele Number Middle Eastern', 'Homozygote Count Middle Eastern',\n",
       "       'Hemizygote Count Middle Eastern',\n",
       "       'Allele Count European (non-Finnish)',\n",
       "       'Allele Number European (non-Finnish)',\n",
       "       'Homozygote Count European (non-Finnish)',\n",
       "       'Hemizygote Count European (non-Finnish)', 'Allele Count Amish',\n",
       "       'Allele Number Amish', 'Homozygote Count Amish',\n",
       "       'Hemizygote Count Amish', 'Allele Count South Asian',\n",
       "       'Allele Number South Asian', 'Homozygote Count South Asian',\n",
       "       'Hemizygote Count South Asian', 'Allele Count Remaining',\n",
       "       'Allele Number Remaining', 'Homozygote Count Remaining',\n",
       "       'Hemizygote Count Remaining'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6e87e20a-c9d1-4361-b0ba-5b63e007a4cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein Consequence\n",
       "p.Leu456Leu          3\n",
       "p.Leu19Leu           3\n",
       "p.Arg161Arg          2\n",
       "p.Trp12GlyfsTer78    2\n",
       "p.Ile118Ile          2\n",
       "                    ..\n",
       "p.Glu333Gly          1\n",
       "p.Glu333Gln          1\n",
       "p.Gln332Arg          1\n",
       "p.Leu331Leu          1\n",
       "p.Ala337Gly          1\n",
       "Name: count, Length: 839, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"Protein Consequence\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "c592097c-7b2a-4f51-9282-ef70e1704d39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'L'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence[455]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d543d42b-efb7-45f8-a80d-a5958131b12d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'I'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence[456]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "30d4569c-4f3a-4e5f-9d37-61bef1a1a855",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'L'"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence[18]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "7ba2ddf3-843e-49a4-aaa6-b530096f93c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'I'"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequence[117]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07c7e19f-f363-4879-891d-16cbb6632fb3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "0bc86532-3fa2-47df-85ce-dff7ba474791",
   "metadata": {},
   "source": [
    "## Its 1 indexed vs biopython 0 indexed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f604579-0a1a-4ae6-b44e-c4cb3b337ba2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
